pkg/ip/cidr.go

```diff
@@ -57,18 +57,6 @@ func ParsePrefixes(cidrs []string) (valid []netip.Prefix, invalid []string, erro
 	return valid, invalid, errors
 }
 
-// AddrToIPNet is a convenience helper to convert a netip.Addr to a *net.IPNet
-// with a mask corresponding to the addresses's bit length.
-func AddrToIPNet(addr netip.Addr) *net.IPNet {
-	if !addr.IsValid() {
-		return nil
-	}
```

Comment:
The behavior between `iputil.AddrToIPNet` and `netipx.AddrIPNet` differs in this case. The former returns nil, the latter an invalid IP. Have you checked if any of the users of `iputil.AddrToIPNet` did a nil-check on the return value?

Reply:
Thanks, good call. I checked all the existing users and non of them seem to assume a `nil` value being returned. It even looks like most users of `iputil.AddrToIPNet` dereference the returned `*net.IPNet` without a `nil` check because they checked the `netip.Addr` passed to it for being valid.

It should thus be safe to make this change.